exchanges/okx/okx.go

```diff
@@ -0,0 +1,4490 @@
+	if err != nil {
+		return nil, err
+	}
+	callbuyvol, err := strconv.ParseFloat(resp[1], 64)
+	if err != nil {
+		return nil, err
+	}
+	callselvol, err := strconv.ParseFloat(resp[2], 64)
+	if err != nil {
+		return nil, err
+	}
+	putbutvol, err := strconv.ParseFloat(resp[3], 64)
+	if err != nil {
+		return nil, err
+	}
+	putsellvol, err := strconv.ParseFloat(resp[4], 64)
+	if err != nil {
+		return nil, err
+	}
+	callblockvol, err := strconv.ParseFloat(resp[5], 64)
+	if err != nil {
+		return nil, err
+	}
+	putblockvol, err := strconv.ParseFloat(resp[6], 64)
+	if err != nil {
+		return nil, err
+	}
+	return &CurrencyTakerFlow{
+		Timestamp:       time.UnixMilli(timestamp),
+		CallBuyVolume:   callbuyvol,
+		CallSellVolume:  callselvol,
+		PutBuyVolume:    putbutvol,
+		PutSellVolume:   putsellvol,
+		CallBlockVolume: callblockvol,
+		PutBlockVolume:  putblockvol,
+	}, nil
+}
+
+// SendHTTPRequest sends an authenticated http request to a desired
+// path with a JSON payload (of present)
+// URL arguments must be in the request path and not as url.URL values
+func (ok *Okx) SendHTTPRequest(ctx context.Context, ep exchange.URL, f request.EndpointLimit, httpMethod, requestPath string, data, result interface{}, authenticated bool) (err error) {
+	endpoint, err := ok.API.Endpoints.GetURL(ep)
+	if err != nil {
+		return err
+	}
+	var intermediary json.RawMessage
+	newRequest := func() (*request.Item, error) {
+		utcTime := time.Now().UTC().Format(time.RFC3339)
+		payload := []byte("")
+
+		if data != nil {
+			payload, err = json.Marshal(data)
+			if err != nil {
+				return nil, err
+			}
+		}
+		path := endpoint + requestPath
+		headers := make(map[string]string)
+		headers["Content-Type"] = "application/json"
+		if authenticated {
+			var creds *account.Credentials
+			creds, err = ok.GetCredentials(ctx)
+			if err != nil {
+				return nil, err
+			}
+			signPath := fmt.Sprintf("/%v%v", okxAPIPath, requestPath)
+			var hmac []byte
+			hmac, err = crypto.GetHMAC(crypto.HashSHA256,
+				[]byte(utcTime+httpMethod+signPath+string(payload)),
+				[]byte(creds.Secret))
+			if err != nil {
+				return nil, err
+			}
+			headers["OK-ACCESS-KEY"] = creds.Key
+			headers["OK-ACCESS-SIGN"] = crypto.Base64Encode(hmac)
+			headers["OK-ACCESS-TIMESTAMP"] = utcTime
+			headers["OK-ACCESS-PASSPHRASE"] = creds.ClientID
+		}
+		return &request.Item{
+			Method:        strings.ToUpper(httpMethod),
+			Path:          path,
+			Headers:       headers,
+			Body:          bytes.NewBuffer(payload),
+			Result:        &intermediary,
+			AuthRequest:   authenticated,
+			Verbose:       ok.Verbose,
+			HTTPDebugging: ok.HTTPDebugging,
+			HTTPRecording: ok.HTTPRecording,
+		}, nil
+	}
+	err = ok.SendPayload(ctx, f, newRequest)
+	if err != nil {
+		return err
+	}
+	type errCap struct {
+		Code string      `json:"code"`
+		Msg  string      `json:"msg"`
+		Data interface{} `json:"data"`
+	}
+	var errMessage errCap
+	errMessage.Data = result
```

Comment:
result needs to have a pointer check or the result is not applied to the calling stack frame. 

```go
rv := reflect.ValueOf(v)
	if rv.Kind() != reflect.Pointer || rv.IsNil() {
    return errSillyBilly
}
```

Reply:
Agreed please change